### PR TITLE
Add test to check that classification models are FX-compatible

### PR DESCRIPTION
**Pull request by fmassa**
This let us know if the FX-based approach for feature extraction from https://github.com/pytorch/vision/pull/3597 would be able to support all classification models in torchvision.

The current approach for feature extraction only work for a subset of the models, and notably doesn't work for inception-style models.

cc @kazhang @jamesr66a for visibility

### CODE REVIEW

File: test/test_models.py
Comment (NicolasHug) [inline test/test_models.py:78]: do the models output classes? otherwise we may need a less strict assertion

**Review by NicolasHug — APPROVED** [review]
small question but LGTM when green!

File: test/test_models.py
Comment (fmassa) [reply test/test_models.py:78]: They output the logits, so there is no max operation performed here.

Comment (NicolasHug) [reply test/test_models.py:78]: Should we use `torch.allclose` then?

Comment (datumbox) [reply test/test_models.py:78]: How is the execution time increased after introducing this? If we have a significant slowdown, it might be worth not reestimating `model(input)` but rather pass it from there one of the other places where we calculate it.

Comment (fmassa) [reply test/test_models.py:78]: > Should we use torch.allclose then?

Sure, we can change it to use `torch.allclose`.

> How much is the execution time increased after introducing this

Good question. I've considered this when writing the tests but went with the same approach that the torchscript tests do to recompute the output.

By comparing the runtimes from this diff and https://github.com/pytorch/vision/pull/3663 (which are based off on the same base), the runtimes don't seem to change significantly

Before https://app.circleci.com/pipelines/github/pytorch/vision/7409/workflows/a7e5719e-7054-4019-bfe9-7c4a5e840900/jobs/511741:
```
9.50s call     test/test_models.py::ModelTester::test_densenet169_cpu
```

this PR https://app.circleci.com/pipelines/github/pytorch/vision/7408/workflows/b9b610f9-6453-4d8e-ac1b-7b7140ebfe26/jobs/511600
```
9.55s call     test/test_models.py::ModelTester::test_densenet161_cpu
```

and the total runtimes don't really seem affected, so I'd say we can keep it like this.

Comment (datumbox) [reply test/test_models.py:78]: > and the total runtimes don't really seem affected, so I'd say we can keep it like this.

Yeap agreed. Since there is no visible slowdown, no point introducing more convoluted code.

**Review by datumbox — APPROVED** [review]
LGTM! Are we merging this or what? :)

**Comment by fmassa** [comment]
Give me some more time before merging this, there is still some things I'd like to check / verify before getting this in.

**Comment by NicolasHug** [comment]
@fmassa I took care of merging with master and making this up-to-date with our new testing infra. LMK if I can help move this forward w.r.t. https://github.com/pytorch/vision/pull/3662#issuecomment-822493617  !

**Comment by fmassa** [comment]
Thanks @NicolasHug !

Let's get this PR merged. I'm not yet sure if we will be adding as constraint that all classification models will be FX-compatible in the future, but at least having this check here will enable us to think about it if it breaks in the future.

**Pull request merged**
